commit 6998d33795b921c6c62dde5a0820242da40e3020
Author: Jason Cooper <jason@lakedaemon.net>
Date:   Wed Oct 6 10:08:00 2010 -0400

    staging: brcm80211: fix 'do not init globals to 0 or NULL'
    
    Signed-off-by: Jason Cooper <jason@lakedaemon.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/brcmfmac/bcmsdh.c
+++ b/drivers/staging/brcm80211/brcmfmac/bcmsdh.c
@@ -43,7 +43,7 @@ struct bcmsdh_info {
 	uint32 sbwad;		/* Save backplane window address */
 };
 /* local copy of bcm sd handler */
-bcmsdh_info_t *l_bcmsdh = NULL;
+bcmsdh_info_t *l_bcmsdh;
 
 #if defined(OOB_INTR_ONLY) && defined(HW_OOB)
 extern int sdioh_enable_hw_oob_intr(void *sdioh, bool enable);
--- a/drivers/staging/brcm80211/brcmfmac/dhd_common.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_common.c
@@ -1363,7 +1363,7 @@ int dhd_preinit_ioctls(dhd_pub_t *dhd)
 
 #ifdef SIMPLE_ISCAN
 uint iscan_thread_id;
-iscan_buf_t *iscan_chain = 0;
+iscan_buf_t *iscan_chain;
 
 iscan_buf_t *dhd_iscan_allocate_buf(dhd_pub_t *dhd, iscan_buf_t **iscanbuf)
 {
--- a/drivers/staging/brcm80211/brcmfmac/dhd_linux.c
+++ b/drivers/staging/brcm80211/brcmfmac/dhd_linux.c
@@ -286,7 +286,7 @@ module_param(dhd_watchdog_ms, uint, 0);
 
 #ifdef DHD_DEBUG
 /* Console poll interval */
-uint dhd_console_ms = 0;
+uint dhd_console_ms;
 module_param(dhd_console_ms, uint, 0);
 #endif				/* DHD_DEBUG */
 
@@ -304,7 +304,7 @@ uint dhd_pkt_filter_enable = TRUE;
 module_param(dhd_pkt_filter_enable, uint, 0);
 
 /*  Pkt filter init setup */
-uint dhd_pkt_filter_init = 0;
+uint dhd_pkt_filter_init;
 module_param(dhd_pkt_filter_init, uint, 0);
 
 /* Pkt filter mode control */
@@ -325,7 +325,7 @@ module_param(dhd_dongle_memsize, int, 0)
 
 /* Contorl fw roaming */
 #ifdef CUSTOMER_HW2
-uint dhd_roam = 0;
+uint dhd_roam;
 #else
 uint dhd_roam = 1;
 #endif
@@ -382,11 +382,11 @@ module_param(dhd_deferred_tx, uint, 0);
 
 #ifdef SDTEST
 /* Echo packet generator (pkts/s) */
-uint dhd_pktgen = 0;
+uint dhd_pktgen;
 module_param(dhd_pktgen, uint, 0);
 
 /* Echo packet len (0 => sawtooth, max 2040) */
-uint dhd_pktgen_len = 0;
+uint dhd_pktgen_len;
 module_param(dhd_pktgen_len, uint, 0);
 #endif
 
--- a/drivers/staging/brcm80211/brcmfmac/linux_osl.c
+++ b/drivers/staging/brcm80211/brcmfmac/linux_osl.c
@@ -123,7 +123,7 @@ static int16 linuxbcmerrormap[] = { 0,
 };
 
 /* Global ASSERT type flag */
-uint32 g_assert_type = 0;
+uint32 g_assert_type;
 
 int osl_error(int bcmerror)
 {
--- a/drivers/staging/brcm80211/brcmfmac/wl_iw.c
+++ b/drivers/staging/brcm80211/brcmfmac/wl_iw.c
@@ -142,7 +142,7 @@ typedef struct iscan_info {
 	wl_iscan_params_t *iscan_ex_params_p;
 	int iscan_ex_param_size;
 } iscan_info_t;
-iscan_info_t *g_iscan = NULL;
+iscan_info_t *g_iscan;
 static void wl_iw_timerfunc(unsigned long data);
 static void wl_iw_set_event_mask(struct net_device *dev);
 static int wl_iw_iscan(iscan_info_t *iscan, wlc_ssid_t *ssid, uint16 action);
